Automattic/wp-calypso
# Billing: Update site-level AddNewPaymentMethod to use PaymentMethodSelector

#### Changes proposed in this Pull Request

This PR updates the site-level "add new payment method" form (previously called `AddNewPaymentMethod` but changed in this PR to `SiteLevelAddNewPaymentMethod` to differentiate it from the account-level one of the same name) to use the `PaymentMethodSelector` component added in https://github.com/Automattic/wp-calypso/pull/49058 but modified to only display the new card option. This replaces `PaymentMethodForm`, which is slowly being deprecated (see https://github.com/Automattic/wp-calypso/issues/48966). The account-level "add new payment method" form was updated in https://github.com/Automattic/wp-calypso/pull/49059.

In the future this will make it easier to modify both forms so that they support a mechanism (see https://github.com/Automattic/wp-calypso/pull/50057) that will allow the customer to choose if they wish to update all their subscriptions at once or not. Since the existing form updates all subscriptions, this change does also.

Before:

<img width="951" alt="Screen Shot 2021-02-09 at 5 47 04 PM" src="https://user-images.githubusercontent.com/2036909/107439100-8c6b9e80-6aff-11eb-8aa6-031f2a30bdd6.png">

After:

<img width="949" alt="Screen Shot 2021-02-09 at 5 47 16 PM" src="https://user-images.githubusercontent.com/2036909/107439116-9097bc00-6aff-11eb-8709-323b602f8b33.png">

A successful addition:

<img width="901" alt="Screen Shot 2021-02-09 at 5 53 03 PM" src="https://user-images.githubusercontent.com/2036909/107439190-b02ee480-6aff-11eb-9540-8bf1777ff7d2.png">

#### Testing instructions

- Visit `/purchases` and click to visit the Payment methods list. **Note: There is another similar form at the account-level Payment methods list but it has already been converted.**
- Click "Add a new payment method".
- Verify that the form loads with no errors and that you see a new credit card form (and no other payment method options).
- Fill out the form and submit it.
- Verify that you are redirected to the payment methods list page and that there's a success message displayed.
- Verify that all your subscriptions now have this new payment method assigned.
